src/components/Signup/Signup.js

```diff
@@ -1,14 +1,26 @@
-import React from "react";
+import React, { useState } from "react";
 import PopupWithForm from "../PopupWithForm/PopupWithForm";
 import { usePopup } from "../../contexts/PopupContext";
-
+import { useStore } from "../../contexts/GlobalContext";
+import { useFormWithValidation } from "../../utils/hooks";
 import "./Signup.css";
 
-function Signup({ isOpen, onClose, onSubmit }) {
+const Signup = () => {
+  const {
+    values,
+    handleChange,
+    resetForm,
+    errors,
+    isValid,
+  } = useFormWithValidation();
   const popupContext = usePopup();
+  const { handleRegister, isLoading } = useStore().UserActions;
+  const [formError, setFormError] = useState({ isError: false, message: "" });
 
   const handleSubmit = (evt) => {
+    const { email, password, username } = values;
     evt.preventDefault();
+    handleRegister({ email, password, name: username });
     popupContext.closeAllPopups();
     popupContext.openPopup("registered");
```

Comment:
if there is an error you can't show that the user is registered


it should be in `then` only